### PR TITLE
(consul) when disabled, only show system health checks

### PR DESCRIPTION
**Pull request by lwander**
@duftler

### CODE REVIEW

File: clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulHealth.groovy [hunk 1]
```diff
@@ -45,6 +45,11 @@ class ConsulHealth extends DiscoveryHealth {
       case CheckResult.Status.passing:
         return HealthState.Up
       case CheckResult.Status.critical:
+        if (isSystemHealth()) {
```
Comment (duftler) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulHealth.groovy]: Please add a comment explaining this logic.

File: clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulNode.groovy [hunk 2]
```diff
@@ -17,7 +17,23 @@
...
   List<ConsulService> services
+
+  boolean isDisabled() {
+    healths.any { health -> // If a check is registered as "_node_maintanence" by consul it's been disabled.
+      health.isSystemHealth()
+    }
+  }
+
+  List<ConsulHealth> getHealths() {
+    if (isDisabled()) { // If the node is disabled, we only return the system health to properly report to Spinnaker that the node should be treated as disabled
```
Comment (duftler) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulNode.groovy]: Missing terminating period.

File: clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulNode.groovy [hunk 3]
```diff
@@ -17,7 +17,23 @@
 package com.netflix.spinnaker.clouddriver.consul.model
 
 class ConsulNode {
-  boolean enabled
+  boolean running
   List<ConsulHealth> healths
   List<ConsulService> services
+
+  boolean isDisabled() {
+    healths.any { health -> // If a check is registered as "_node_maintanence" by consul it's been disabled.
```
Comment (duftler) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulNode.groovy]: Please add a comma after 'consul'.

**Comment by duftler** [comment]
LGTM after addressing comments.

File: clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulHealth.groovy (hunk 1 above)
Comment (lwander) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/model/ConsulHealth.groovy]: Added an explanation on the next line.

**Pull request merged**